android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt

```diff
@@ -287,54 +291,108 @@ constructor(
+        val practitionerId =
+          practitionerDetails.fhirPractitionerDetails?.practitionerId.valueToString()
+        val locationHierarchies =
+          practitionerDetails.fhirPractitionerDetails?.locationHierarchyList ?: listOf()
+
+        val careTeamIds =
+          withContext(dispatcherProvider.io()) {
+            defaultRepository.createRemote(false, *careTeams.toTypedArray()).run {
+              careTeams.map { it.id.extractLogicalIdUuid() }
+            }
           }
-        }
-      val organizationIds =
-        withContext(dispatcherProvider.io()) {
-          defaultRepository.createRemote(false, *organizations.toTypedArray()).run {
-            organizations.map { it.id.extractLogicalIdUuid() }
+        val organizationIds =
+          withContext(dispatcherProvider.io()) {
+            defaultRepository.createRemote(false, *organizations.toTypedArray()).run {
+              organizations.map { it.id.extractLogicalIdUuid() }
+            }
           }
-        }
-      val locationIds =
-        withContext(dispatcherProvider.io()) {
-          defaultRepository.createRemote(false, *locations.toTypedArray()).run {
-            locations.map { it.id.extractLogicalIdUuid() }
+        val locationIds =
+          withContext(dispatcherProvider.io()) {
+            defaultRepository.createRemote(false, *locations.toTypedArray()).run {
+              locations.map { it.id.extractLogicalIdUuid() }
+            }
           }
-        }
 
-      sharedPreferences.write(
-        key = SharedPreferenceKey.PRACTITIONER_ID.name,
-        value = practitionerDetails.fhirPractitionerDetails?.id
-      )
-
-      sharedPreferences.write(SharedPreferenceKey.PRACTITIONER_DETAILS.name, practitionerDetails)
-      sharedPreferences.write(ResourceType.CareTeam.name, careTeamIds)
-      sharedPreferences.write(ResourceType.Organization.name, organizationIds)
-      sharedPreferences.write(ResourceType.Location.name, locationIds)
-      sharedPreferences.write(
-        SharedPreferenceKey.PRACTITIONER_LOCATION_HIERARCHIES.name,
-        locationHierarchies
-      )
+        defaultRepository.createRemote(false, *practitioners.toTypedArray())
+        practitionerDetails.fhirPractitionerDetails?.groups?.toTypedArray()?.let {
+          defaultRepository.createRemote(false, *it)
+        }
+        practitionerDetails.fhirPractitionerDetails?.practitionerRoles?.toTypedArray()?.let {
+          defaultRepository.createRemote(false, *it)
+        }
+        practitionerDetails.fhirPractitionerDetails?.organizationAffiliations?.toTypedArray()?.let {
+          defaultRepository.createRemote(false, *it)
+        }
 
+        if (practitionerId.isNotEmpty()) {
+          writePractitionerDetailsToShredPref(
+            practitionerDetails,
+            careTeamIds,
+            organizationIds,
+            locationIds,
+            locationHierarchies
```

Comment:
I think before we update the shared prefs we should also add the same check we do to verify that they are the currently logged in user